System/Directory.hs

```diff
@@ -401,7 +401,9 @@ removeDirectory = removePathInternal True
 -- together with its contents and subdirectories. Within this directory,
 -- symbolic links are removed without affecting their targets.
 --
--- On Windows, the operation fails if /dir/ is a directory symbolic link.
+-- On Windows, the operation fails if /dir/ is a directory symbolic link. 
+-- Rarely, the operation may also fail on Windows if a directory is removed 
+-- immediately after a child directory.
```

Comment:
Can you elaborate on what "directory is removed immediately after a child directory" means? I'm sense like there's a verb missing after "child directory", but maybe I'm misinterpreting the sentence.

Reply:
How about:

`Rarely, the operation may also fail on Windows if a directory is removed immediately after a all of its children directories.`

As for the official documentation, the `win32` library uses `RemoveDirectory`, which is an alias for [`RemoveDirectoryA`](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-removedirectorya). The documentation states:

> The RemoveDirectory function marks a directory for deletion on close. Therefore, the directory is not removed until the last handle to the directory is closed.

Therefore, calling `RemoveDirectory` twice (first on child, then parent) may result in a "Directory not empty" error on the second call. It's not stated *explicitly*, though, so I don't know how helpful it would be to include a link the official docs.

Reply:
Thanks for the explanation. There seems to be a fair bit of nuance in this failure mode. In particular, I read failures could also be caused by external programs like antivirus software, perhaps there are multiple root causes?

Maybe it would be simpler to not explain the root cause and just link to this GitHub PR?

    This operation is reported to be flaky on Windows so retry logic may be advisable.
    See: https://github.com/haskell/directory/pull/108

Reply:
Yes, this is better. Latest commit resolves this.